src/aq/Geometry.cc

```diff
@@ -93,6 +109,48 @@ GeometryIterator Geometry::end() const {
   return GeometryIterator(*this, nx*ny+1);
 }
 // -------------------------------------------------------------------------------------------------
+void Geometry::latlon(std::vector<double> & lats, std::vector<double> & lons,
+                      const bool halo) const {
+  if (halo_ > 0) {
+    const atlas::functionspace::StructuredColumns * fspace;
+    if (halo) {
+      fspace = atlasFunctionSpace_.get();
+    } else {
+      fspace = atlasFunctionSpaceNoHalo_.get();
+    }
+    const auto lonlat = atlas::array::make_view<double, 2>(fspace->lonlat());
+    const size_t npts = fspace->size();
+    lats.resize(npts);
+    lons.resize(npts);
+    for (size_t jj = 0; jj < npts; ++jj) {
+      lats[jj] = lonlat(jj, 1);
+      lons[jj] = lonlat(jj, 0);
+    }
+  } else {
+    /* If the fields have no halo, the locations have to be only on proc 0, therefore latlon
+       is provided only there */
```

Comment:
That's a cool trick! Before I saw this, I was going to say it wouldn't be possible to compute H(x) on just task 0, and avoid distributing the obs across all processors.

Reply:
I am prouder than a rooster if Anna admires my tricks. I keep an eye on future different uses of latlon in order to avoid side effects.
I've written to Willem asking for an atlas interpolation for Regional StructuredColumns to PointCloud (and its adjoint) with fortran API, so that I'll be able to switch to a distributed surface interpolation.

Thank you for the warning on JCSDA-internal/oops#1700. Sice I try - puffing and panting - to follow the notification mails flying around as stardust, I had already prepared the lons and lats methods, but this kind of warnings and indications (instead just of failing tests) it's really what we dreamed of. 

Andrea